src/include/txn.i

```diff
@@ -746,6 +746,23 @@ __wt_txn_upd_visible(WT_SESSION_IMPL *session, WT_UPDATE *upd)
     return (__wt_txn_upd_visible_type(session, upd) == WT_VISIBLE_TRUE);
 }
 
+/*
+ * __upd_alloc_tombstone --
+ *     Allocate a tombstone update at a given transaction id and timestamp.
+ */
+static inline int
+__upd_alloc_tombstone(
+  WT_SESSION_IMPL *session, WT_UPDATE **updp, uint64_t txnid, wt_timestamp_t start_ts)
+{
+    size_t size;
+
+    WT_RET(__wt_update_alloc(session, NULL, updp, &size, WT_UPDATE_TOMBSTONE));
+    (*updp)->txnid = txnid;
+    (*updp)->durable_ts = (*updp)->start_ts = start_ts;
+    F_SET(*updp, WT_UPDATE_RESTORED_FROM_DISK);
+    return (0);
+}
+
```

Comment:
Reconciliation already has `__tombstone_update_alloc()`, and `__wt_find_hs_upd()`, `__wt_txn_read()`, `__rec_append_orig_value()`, `__wt_rec_upd_select()` all allocate a `WT_UPDATE` structure and immediately fill out the ts/txn information.

I've wondered for awhile if we need a variant of `__wt_update_alloc()` that takes timestamp/txn information, maybe that's a better way forward? It might help us avoid bugs by requiring we ensure all the ts/txn fields are filled in?

Reply:
Hi, I've made a ticket for the refactor so I think we can address it then: WT-5470.
Most of these usages should disappear.